tests/test_userhistory.py

```diff
@@ -1,33 +1,44 @@
-""" Tests for user history class """
+''' Tests for user history class '''
 import __init__
 from userhistory import UserHistory
 
 def test_userhistory_init():
-    """Tests user history init"""
-    # specifying a short date range prevents unnecessarily pulling down
-    # a very large number of revisions - saves us time on testing
-    user_history = UserHistory("Jimbo Wales",
-                               startyear=2022, startmonth=1, startday=1,
-                               endyear=2022, endmonth=1, endday=1)
+    '''Tests user history init'''
+    user_history = UserHistory("QuicoleJR")
 
-    assert user_history.user == "Jimbo Wales"
+    assert user_history.user == "QuicoleJR"
+    assert user_history.tags is None
+    assert user_history.keyword is None
+    assert len(user_history.revisions) > 0
```

Comment:
I like this change, it's good to check on the None values and that revisions were pulled down and I feel that it makes sense here as opposed to getting its own test function - this is more a subroutine of __init__ than an independent behavior.

Regarding testing for the presence of revisions that should have been pulled down from the external API and included in user_history.revisions, I have been giving our test suite more thought lately and I have a lot of thoughts that tie in here. This would be an excellent time for me to learn how to use the github discussions feature. I'll try and do that after this review and maybe link back to this code to anchor a future conversation about the suite in general.

Reply:
One of the main thoughts that I have is with regards to the default behavior we have right now, which pulls down every revision ever for a given user or article if no date range is specified. I think we should change this and will elaborate in a conversation, but for now I want to suggest that tests should include some level of start and end specificity to limit this testing overhead.

Additionally, when our default behavior changes, tests should specify consistent start and end datetimes so that we can guarantee that we get the expected responses back that allow the tests to pass, so I think this change is worth implementing either way. Unfortunately this does also muddy the filter tests a little bit since they would all then depend on the date/time code outside the filters they're supposed to be testing. We could potentially mock this in some way but this is starting to get highly complicated - I recall Daniel saying in a lecture that we should expect to spend as much as 2-3 hours writing tests for each hour we spend writing code, and I think this is a part of what he was talking about. If you have a bit of time today to implement this in the context of this PR that would be awesome, but I understand if it can't happen today and am happy to merge this as-is for the time being with the intent that we address this in all tests some time this last sprint. I'll get into it more in a conversation but I think we should spend significant time improving and building out our test suite, not only for the sake of the project overall but also as an exercise that's pretty relevant to the course material.

Reply:
Updated to use timestamps that are consistent between all user history tests, we can further discuss implementing mocking in our tests